src/form.js

```diff
@@ -274,23 +276,63 @@ define([
          * @returns - The `id` of the added or already-existing instance.
          */
         addInstanceIfNotExists: function (attrs, mug, property) {
-            if (attrs.src === null || _.isUndefined(attrs.src)) {
-                // no ref counting for instances without a `src` since they cannot be dropped
-                this.instanceMetadata.push(InstanceMetadata({
-                    src: attrs.src,
-                    id: attrs.id
-                }));
-                return attrs.id;
+            function getUniqueId(id, ids) {
+                var temp = (id || "data") + "-",
+                    num = 1;
+                while (!id || ids.hasOwnProperty(id)) {
+                    id = temp + num;
+                    num++;
+                }
+                return id;
             }
-
-            var meta = _.find(this.instanceMetadata, function (m) {
+            var meta;
+            if (attrs.src) {
+                meta = _.find(this.instanceMetadata, function (m) {
                     return m.attributes.src === attrs.src;
                 });
+                if (!meta) {
+                    // attrs.src not found, try to find by id
+                    var ids = _.indexBy(this.instanceMetadata, function (m) {
```

Comment:
`_.indexBy` was introduced in 1.5.2. HQ serves up 1.4.4.

This actually works right now because `underscore` is in our local-deps. I think we should probably change the version to 1.4.4 in bower and move it from local-deps to global-deps, so that we aren't loading two versions of _